numba/core/interpreter.py

```diff
@@ -877,20 +877,23 @@ def op_CALL_FUNCTION_EX(self, inst, func, vararg, res):
             expr = ir.Expr.call(func, [], [], loc=self.loc, vararg=vararg)
             self.store(expr, res)
 
-    def _build_tuple_unpack(self, inst, tuples, temps):
+    def _build_tuple_unpack(self, inst, tuples, temps, is_assign):
         first = self.get(tuples[0])
-        for other, tmp in zip(map(self.get, tuples[1:]), temps):
-            out = ir.Expr.binop(fn=operator.add, lhs=first, rhs=other,
-                                loc=self.loc)
-            self.store(out, tmp)
-            first = self.get(tmp)
+        if is_assign: # it's assign-like, just store through
+            self.store(first, temps[0])
```

Comment:
There needs to be a convert to tuple.

```

In [5]: @njit
   ...: def foo(x):
   ...:     y = (*x,)
   ...:     return y
   ...:

In [6]: foo([1, 2])

Out[6]: [1, 2]
```

Reply:
Is there actually enough information available to handle this?
```python
In [1]: import dis

In [2]: def foo(x):
   ...:     return (*x,)
   ...: 

In [3]: foo((1,2))
Out[3]: (1, 2)

In [4]: foo([1,2])
Out[4]: (1, 2)

In [5]: dis.dis(foo)
  2           0 LOAD_FAST                0 (x)
              2 BUILD_TUPLE_UNPACK       1
              4 RETURN_VALUE

```
the return type ends up depending on the list size.

Reply:
Should we raise UnsupportedError/TypeError instead when the `x` in `(*x,)` is not a `tuple`?

Reply:
I think it has to be an error if it's not a `tuple`. From OOB conversation, going try with emitting an intrinsic to handle this, can't do anything from the bytecode interpreter as there's nothing unique about the tuple vs. anything else situation.

Reply:
Managed to get this to work in 28bdbca.